### PR TITLE
Add HTTP basic auth

### PR DESCRIPTION
**Pull request by jjbohn**
Long story short, Net http was removing the HTTP basic auth, which is why the requests stopped going through brazil when I added auth to the url.

### CODE REVIEW

File: test/test_logger.rb [hunk 1]
```diff
@@ -9,10 +9,12 @@ def test_standard_request
       application: "brazil",
       environment: "development",
       host: "https://elasticsearch.dev:9200",
+      username: "foo",
+      password: "bar",
```
Comment (abepetrillo) [inline test/test_logger.rb]: Trailing `,` here?

Comment (jjbohn) [inline test/test_logger.rb]: I usually do trailing commas in multi line arrays and hashes. It makes it so if I add another element, when you diff in git, there's no removal, it's just a single additon, as opposed to two additions and a deletion.

Comment (jjbohn) [inline test/test_logger.rb]: It's why host is shown as an unchanged line.

Comment (abepetrillo) [inline test/test_logger.rb]: Seems weird to me but ok.

**Comment by abepetrillo** [comment]
:+1:

File: test/test_logger.rb (hunk 1 above)
Comment (jjbohn) [inline test/test_logger.rb]: It's a pretty religious thing. For example: https://github.com/bbatsov/ruby-style-guide/issues/273

That said, I'm a fan. Thoughtbot's style guide has it and I usually recommend it. Just is nicer to me in general.

**Pull request merged**